### ISSUE TITLE
Freeleech tokens

### ISSUE THREAD

**Issue by rubdos** [Jackett/Jackett#2232]
Certain trackers (AR et al.) allow for the use of "freeleech tokens" on certain backlog torrents. Many of these trackers offer *so many* of these tokens that it's almost guaranteed to have them at all times.

It would be nice if one could configure Jackett to return freeleech-token-enabled Torrent urls, or have them configurable in the client software (radarr/sonarr), e.g. a checkbox "use freeleech tokens when available".

E.g. for AR, the difference in Torrent url is "&usetoken=1".

**Comment by kaso17** [comment]
Having a option to append &usetoken=1 wouldn't be a problem.
Can you please test what happens when you add &usetoken=1 to torrents which don't support FL tokens and what happens if you use &usetoken=1 but have no more tokens available?
If you still get a valid torrent file it would be easy to implement.

**Comment by rubdos** [comment]
> Can you please test what happens when you add &usetoken=1 to torrents which don't support FL tokens and what happens if you use &usetoken=1 but have no more tokens available?
> If you still get a valid torrent file it would be easy to implement.

I know I get a valid torrent in both cases. This is for Gazelle based trackers (APL,AR). When you use `usetoken=1` on a global freeleech torrent, no tokens are subtracted, and iirc, nothing happens if you run out.

**Comment by kaso17** [comment]
implemented via 29f111aec40ef9defaf8afd513b8ad4ae3a02c71

**Comment by rubdos** [comment]
I'm going with a very vague bug statement here: "it doesn't work". I checked the "use freeleech tokens" checkbox. How do I debug this? (Make a new issue?)

**Comment by kaso17** [comment]
please enable enhanced logged and try downloading a torrent, you should see the download link (including the usetoken part in the log)

**Comment by karelvanhecke** [comment]
@kaso17 I tried it with AR and it uses usetoken=0 in the URL, even when enabled.

**Comment by rubdos** [comment]
> @kaso17 I tried it with AR and it uses usetoken=0 in the URL, even when enabled.

Ha, thanks for trying this. I didn't have much time to test lately.

**Comment by kaso17** [comment]
Working fine for me (uses &usetoken=1). Can you please share an enhanced log (-t and -l command line option) while doing the indexer configuration (enable freeleech tokens), a search and then the download?

**Comment by karelvanhecke** [comment]
@kaso17 I readded AlphaRatio and that fixed the issue. I can now confirm tokens are being used. This issue can be closed.

**Comment by rubdos** [comment]
> Can you please test what happens when you add &usetoken=1 to torrents which don't support FL tokens and what happens if you use &usetoken=1 but have no more tokens available?

Apollo sadly returns

```html
<p>This torrent is too large. Please use the regular DL link.</p>
```

when requesting freeleech on >2GB. :'(

**Comment by kaso17** [comment]
@rubdos do you know if other gazelle trackers return the same error for large torrents or is it just Apollo?

**Comment by kaso17** [comment]
fixed via fc3d8d1aec83315b2d58ade10392949ab3e322d5
